client/src/context/useBoardContext.tsx

```diff
@@ -33,19 +33,17 @@ export const BoardProvider: FunctionComponent = ({ children }): JSX.Element => {
     console.log(currentBoard);
   };
 
-  const fetchBoard = async (user: User, index: number) => {
-    if (!user.boards || user.boards.length <= index) return;
-    const id = user.boards[index];
-    await getBoard(id).then((data: UpdateBoardApiData) => {
-      if (data.success) setCurrentBoard(data.success);
-    });
+  const fetchBoard = async (id: string) => {
+    const data: UpdateBoardApiData = await getBoard(id);
+    if (data.success) setCurrentBoard(data.success);
+    else throw new Error(data.error ? data.error.message : 'An unknown error occurred');
```

Comment:
Would we want to crash the whole app for the user if they couldn't get their board? Probably some better error handling could be done here?